### ISSUE TITLE
HC12 Batching only executes first operation

### ISSUE THREAD

**Issue by MarcoCunha0** [ChilliCream/hotchocolate#4646]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I noticed that in HC12, when i make a batch of requests i am only getting the result to the first operation. This issue was already reported in [slack](https://hotchocolategraphql.slack.com/archives/CD9TNKT8T/p1638459135317900) by Chris Harris. 

According to Chris, this issue seems to be localized in the MultiPartResponseStreamSerializer and his proposal worked for me, which is using: 
```
services.AddHttpResultSerializer(
   batchSerialization: HttpResultSerialization.JsonArray
);
```

### Steps to reproduce

Make a request with a batch of operations:

```
[
	{
		"query": "query DummyQuery1 { buildVersion }",
		"operationName": "DummyQuery1"
	},
    {
		"query": "query DummyQuery2 { buildVersion }",
		"operationName": "DummyQuery2"
	}
]
```
Response:

```
---
Content-Type: application/json; charset=utf-8

{"data":{"buildVersion":"1.0.22011.112"}}
-----
```

### Product

Hot Chocolate

### Version

12.4.1

**Comment by michaelstaib** [comment]
We have started work on batching.

**Comment by Helsing112** [comment]
This bug is still present in 12.7.0. 
Im unsure how to give further detail of the bug. I created a fresh project following the getting started section of the docs and here the bug is present aswell.

**Comment by katyalh** [comment]
@michaelstaib this issue is reproducible in v12.12.1, for now the workaround is to use `HttpResultSerialization.JsonArray` but by using it we lose the incremental delivery over HTTP. Any suggestions there?

**Comment by nelsonprsousa** [comment]
I think @michaelstaib fixed this in v13

**Comment by katyalh** [comment]
@nelsonprsousa @michaelstaib by any chance there will be a release including a fix for this issue against v12 or do we need to wait v13 release?

**Comment by katyalh** [comment]
> I think @michaelstaib fixed this in v13

I have tried the same thing v13.0.0-preview.28, this issue is reproducible there as well

**Comment by michaelstaib** [comment]
Not sure yet where we put this. We will fix it. But have no clue when yet. I will reevaluate this at the end of this month. At the moment everyone in the team busy on other topics.

**Comment by michaelstaib** [comment]
This is now fixed and will be integrated into 12.14.0-preview.1
